### ISSUE TITLE
Can't compile @inertiajs/inertia-vue/src/link.js (Unexpected token)

### ISSUE THREAD

**Issue by jamesburrow23** [inertiajs/inertia#414]
### Versions:

- `@inertiajs/inertia` version: 0.8.2
- `@inertiajs/inertia-vue` version: 0.5.4
- `npm v7.3.0`
- `node v15.5.1`
- `ubuntu 20.04`
- `laravel-mix v5.0.9`

### Describe the problem:
When running `npm run dev` I experience the error below:

```
ERROR in ./node_modules/@inertiajs/inertia-vue/src/link.js 83:50
Module parse failed: Unexpected token (83:50)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|               replace: props.replace,
|               preserveScroll: props.preserveScroll,
>               preserveState: props.preserveState ?? (method !== 'get'),
|               only: props.only,
|               headers: props.headers,
 @ ./node_modules/@inertiajs/inertia-vue/src/app.js 2:0-25 85:33-37
```

I'd think it would compile without errors
### Steps to reproduce:
Not exactly sure, I'd assume you could install the same versions I'm running and try to compile with the same npm and node and see what happens?

**Comment by sebastiandedeyne** [comment]
Could you share some more code? (Webpack configuration, import statements,…)

The underlying problem is that your build process is importing from the packages `src` folder instead of `dist`.

**Comment by claudiodekker** [comment]
This has to do with that you're compiling against the wrong EcmaScript (Javascript) version, as the [`??` (Nullish Coalescing Operator)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_Coalescing_Operator) that this is failing on [was only added in `ES2020`](https://github.com/tc39/proposals/blob/master/finished-proposals.md)

That said, in normal situations you indeed shouldn't have a need to be compiling the package yourself, and should instead be using the `dist` variants.

**Comment by jamesburrow23** [comment]
@sebastiandedeyne I'm not sure why it would be using the src folder, I haven't changed anything in my build process since upgrading to v 0.5.4 of inertia-vue.  i was previously on v4 which doesn't have that `??` operator.  

My use of inertia in my vue app is pretty much from the docs with a little bit of difference in the resolveComponent example, could that be the culprit?

```
import { App, plugin } from '@inertiajs/inertia-vue'
import Vue from 'vue'

Vue.use(plugin)

const el = document.getElementById('app')

new Vue({
  render: h => h(App, {
    props: {
      initialPage: JSON.parse(el.dataset.page),
      resolveComponent: name => import(`./views/${name}.vue`).then(module => module.default);
    },
  }),
}).$mount(el)
```

**Comment by jamesburrow23** [comment]
I tried it without the module.default import and it still seems to be pulling from src instead of dist

**Comment by claudiodekker** [comment]
Are you perhaps doing something special in your webpack / laravel mix config file? That's the only thing that I can think of that would cause this.

**Comment by reinink** [comment]
I wonder if this is related to #394 at all. 🤔

**Comment by jamesburrow23** [comment]
@claudiodekker I tried with a plain vanilla webpack.config.js file too to try to isolate the problem but was still facing the issue.  I even replaced my app.js file with what the docs have only importing vue and inertia and it still wasn't working.

**Comment by jamesburrow23** [comment]
I wasn't able to replicate this issue in a brand new install of laravel + inertia so somewhere my application is just behaving weirdly.  Thanks for the feedback

**Comment by jamesburrow23** [comment]
In case anyone stumbles across this and happens to be as confused as I was....upon further inspection @sebastiandedeyne was exactly right.  I was attempting to compile from src, I was so confused because I would never knowingly do this, right?

Well PHPStorm thought it was being really helpful when it randomly inserted this line at the beginning of one of my vue partial components `import Inertia from "@inertiajs/inertia-vue/src/app";`  Apparently it's been there since June and I never noticed because there wasn't any issues compiling from src until now.

Long story short - assume nothing, check everything, stay calm and carry on :triumph:
